x-pack/plugins/ingest_manager/server/services/agents/acks.ts

```diff
@@ -29,8 +35,18 @@ export async function acknowledgeAgentActions(
 ): Promise<AgentAction[]> {
   const now = new Date().toISOString();
 
+  const actionIds = agentEvents
+    .map(event => event.action_id)
+    .filter(actionId => actionId !== undefined);
+  const actions = await getAgentActionByIds(soClient, actionIds as string[]);
+  for (const action of actions) {
+    if (action.agent_id !== agent.id) {
+      throw Boom.badRequest(`${action.id} cannot be allowed by this agent`);
+    }
+  }
+
   const agentActionMap: Map<string, AgentAction> = new Map(
-    agent.actions.map(agentAction => [agentAction.id, agentAction])
+    actions.map(agentAction => [agentAction.id, agentAction])
   );
```

Comment:
I think that because of these lines we may want to revise, change or delete these line https://github.com/elastic/kibana/pull/62137/files#diff-aa06caf88e4052fdf14b980fb52b8e60R58-R66 . If we get past here can we say all the actions are matched. And we may want to do this first before we search for actions? https://github.com/elastic/kibana/pull/62137/files#diff-aa06caf88e4052fdf14b980fb52b8e60L38-R57